src/plugins/cosmos/components/modals/Staking/views/Stake.tsx

```diff
@@ -87,6 +93,19 @@ export const Stake = ({ assetId, apr }: StakeProps) => {
 
   const translate = useTranslate()
 
+  const chainAdapterManager = useChainAdapters()
+  const adapter = chainAdapterManager.byChain(asset.chain) as cosmossdk.cosmos.ChainAdapter
+
+  useEffect(() => {
+    ;(async () => {
+      const averageTxFee = await adapter.getFeeData({})
+
+      const txFees = getFormFees(averageTxFee, asset.precision, marketData.price)
+
+      setaverageTxFee(txFees[chainAdapters.FeeDataKey.Average])
+    })()
+  }, [adapter, asset.precision, marketData.price])
+
```

Comment:
Self-review: Since we now have the same duplicated logic in `<Stake />`/`<StakeConfirm />` for getting the fees, we could abstract it into form state, to avoid repeating ourselves.